src/Hosting/Hosting/src/Internal/HostingMetrics.cs

```diff
@@ -0,0 +1,168 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Metrics;
+
+namespace Microsoft.AspNetCore.Hosting;
+
+internal sealed class HostingMetrics : IDisposable
+{
+    public const string MeterName = "Microsoft.AspNetCore.Hosting";
+
+    private readonly Meter _meter;
+    private readonly UpDownCounter<long> _currentRequestsCounter;
+    private readonly Histogram<double> _requestDuration;
+
+    public HostingMetrics(IMeterFactory meterFactory)
+    {
+        _meter = meterFactory.CreateMeter(MeterName);
+
+        _currentRequestsCounter = _meter.CreateUpDownCounter<long>(
+            "current-requests",
+            description: "Current requests");
+
+        _requestDuration = _meter.CreateHistogram<double>(
+            "request-duration",
```

Comment:
We should put `unit` value here, and I suggest that we use `second` instead of `millisecond`.

Reply:
@CodeBlanch @noahfalk @tarekgh FYI.

Reply:
Why seconds? I followed what OpenTelemetry does for this counter. They use `ms` - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#metric-httpserverduration

Reply:
Yes, its unclear why we'd use seconds here.

Reply:
Check https://github.com/open-telemetry/opentelemetry-specification/issues/2977#issuecomment-1489150771.

Reply:
> Why seconds? I followed what OpenTelemetry does for this counter. They use `ms` - https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#metric-httpserverduration

This entire doc/spec is marked as Experimental and I'm 99.99% confident that it will change in a breaking way.

![image](https://user-images.githubusercontent.com/17327289/229682271-e09b08c9-2b10-42fe-9ac7-27a827c8b0ca.png)

Reply:
I also think it is _okay_ for .NET components to stick to their own schema/convention (rather than giving an illusion that .NET is trying to be compliant with OpenTelemetry semantic convention) since .NET is unlikely to be tight to the lifecycle of the OpenTelemetry semantic convention.

My suggestion for using seconds instead of milliseconds is more coming from the Prometheus ecosystem (which seems to be the de-facto standard for metrics).

Reply:
If the metrics ecosystem standard is seconds, then I don't have a problem with seconds. There are some .NET logs that report ms, e.g. `Executed page /Privacy in 74.5188ms` but I don't think we need to stick to it.

When do you expect that semantic OTel specs to be locked down? We can easily react to changes until about August.

Reply:
I think the spec will move to Stable before August.

Reply:
Why does it matter to you? double should have enough precision to capture good data even if the typical request duration is still small. You can convert it to milliseconds when presented to the user.

Reply:
> Why does it matter to you? double should have enough precision to capture good data even if the typical request duration is still small. You can convert it to milliseconds when presented to the user.

+1 https://github.com/open-telemetry/opentelemetry-specification/issues/2977#issuecomment-1324958938

Reply:
> Why does it matter to you? double should have enough precision to capture good data even if the typical request duration is still small. You can convert it to milliseconds when presented to the user.

Where do we convert it? `OpenTelemetry` doesn't provide any hook to convert metrics unit after it's collected today. the only place for conversion than is going to be backend when actually creating dashboards and alerts one needs to repeat this step for every single dashboard and alert remembering to do it specifically for this metric while all of their other metrics are in milliseconds. So, this will then lead us to continue having our own implementation.

Also, you can use the same argument for converting from milliseconds to seconds.

Reply:
> repeat this step for every single dashboard and alert

Are you expecting that these metrics will be a drop-in replacement and most pre-existing dashboards/alerts were going to automatically work? I was not expecting that to be the case given the name changes and the introduction of attributes. I assumed we'd need some updated default dashboards+alerts to go along with this and teams would need to opt into using them.

> while all of their other metrics are in milliseconds

It sounds like the OTel TC group assessed that the majority of the industry is already using seconds as the default time measurement unit and having OTel endorse seconds as their standard is going to further push people in that direction. I think we'd be better off embracing the industry trend rather than trying to resist it. If teams already need to do a little bit of migration work to start using these metrics it feels like a pretty natural time to handle the change of unit as well.